src/utils/fileOperations.ts

```diff
@@ -0,0 +1,67 @@
+import { promises as fs } from "fs";
+import path from "path";
+
+async function readFileContent(filePath: string): Promise<string> {
+  try {
+    const data = await fs.readFile(filePath, "utf-8");
+    return data;
+  } catch (error) {
+    console.error("Error reading file:", error);
+    throw error;
+  }
+}
+
+function convertStringToJson(data: string): any {
+  try {
+    return JSON.parse(data);
+  } catch (error) {
+    console.error("Error parsing JSON:", error);
+    throw error;
+  }
+}
+
+async function writeToJsonFile(filePath: string, data: string): Promise<void> {
+  try {
+    const jsonData = convertStringToJson(data);
+    await fs.writeFile(
+      filePath + ".json",
+      JSON.stringify(jsonData, null, 2),
+      "utf-8"
+    );
+    console.log(`Response written successfully to ${filePath}.json`);
+  } catch (error) {
+    console.error("Error while saving as JSON:", error);
+    console.log("Trying to save as text instead...");
+    writeToTextFile(filePath, data);
+  }
+}
+
+async function writeToTextFile(filePath: string, data: string): Promise<void> {
+  try {
+    await fs.writeFile(filePath + ".txt", data, "utf-8");
+    console.log(`Response written successfully to ${filePath}.txt`);
+  } catch (error) {
+    console.error("Error writing file:", error);
+    throw error;
+  }
+}
+
+async function convertToAnkiDeck(filePath: string): Promise<string> {
```

Comment:
The name of this function is misleading, as it's not creating an Anki deck per se, which is an `.apkg` file.

```suggestion
async function createTextImportableToAnki(filePath: string): Promise<string> {
```

Reply:
Okay, it will be more precise indeed.